internal/authn/mocks/mock_tracing_server.go

```diff
@@ -0,0 +1,48 @@
+package mocks
```

Comment:
Since the mocked interface contained herein is not related to `authn`, can we please put these in an alternative package, maybe simply `internal/mocks` or `internal/telemetry/mocks`?

Reply:
Also, do we need this mocked interface if we remove the test above?

Related to https://github.com/openfga/openfga/pull/822/files#r1244374750

Reply:
Yes the mock is still useful because the test is still useful :D